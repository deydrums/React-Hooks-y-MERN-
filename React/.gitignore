# proyectos
/00-proyectos