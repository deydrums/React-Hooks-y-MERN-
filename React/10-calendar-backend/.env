PORT=80
DB_CNN=mongodb+srv://deydrums:7VaOymNqlPNdUn12@cluster0.id2bg.mongodb.net/CalendarApp
SECRET_JWT_SEED=Esto-Es-una-PalAbRa-S3kr2t4-j3j3j3