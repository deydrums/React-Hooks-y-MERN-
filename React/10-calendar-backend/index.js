const express = require('express');
require('dotenv').config();
//Ejecutar express
var app = express();

//Directorio publico
app.use(express.static('public'));

//Rutas
//Reescribir rutas
app.use('/api/auth', require('./routes/auth'));

//TODO: CRUD: Eventos

//Escuchar peticiones
app.listen(process.env.PORT,()=>{
    console.log(`Servidor corriendo en puerto ${process.env.PORT}`)
});