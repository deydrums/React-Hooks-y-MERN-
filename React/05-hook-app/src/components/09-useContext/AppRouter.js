import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { ErrorComponent } from './ErrorComponent';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar></NavBar>
                <Switch>
                    <Route exact path="/about" component={AboutScreen}/>
                    <Route exact path="/login" component={LoginScreen}/>
                    <Route exact path="/" component={HomeScreen}/>
                    <Route exact path="*" component={ErrorComponent}/>
                    {/* <Redirect to="/"/> */}
                </Switch>
            </div>
        </Router>
    )
}
