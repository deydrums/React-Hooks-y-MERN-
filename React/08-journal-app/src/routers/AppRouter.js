import firebase from 'firebase';
import React, { useEffect, useState } from 'react'
import { useDispatch } from 'react-redux';
import {
    HashRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";
import { login } from '../actions/auth';
import { LoadingScreen } from '../components/auth/LoadingScreen';
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';

export const AppRouter = () => {
    const dispatch = useDispatch();
    const [cheking, setCheking] = useState(true);
    const [isLoggedIn, setIsLoggedIn] = useState(false);



    useEffect(() => {
        firebase.auth().onAuthStateChanged((user) => {
            if(user?.uid){
                dispatch(login(user.uid, user.displayName));
                setIsLoggedIn(true);
            }else{
                setIsLoggedIn(false);
            };
            setCheking(false);
        });
        
    }, [dispatch, setCheking, setIsLoggedIn]);

    if(cheking){
        return (
            <LoadingScreen></LoadingScreen>
        )
    }

    return (
        <Router>
            <Switch>
                <Route path="/auth" component ={AuthRouter}/>
                <Route exact path="/" component ={JournalScreen}/>
                <Redirect to = "/auth/login"/>
            </Switch>
        </Router>
    )
}
