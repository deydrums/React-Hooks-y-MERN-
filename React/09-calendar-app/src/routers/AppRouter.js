import React from 'react';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import {
  HashRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch>
                    <Route exact path="/login" component={LoginScreen}/>
                    <Route exact path="/" component={CalendarScreen}/>
                </Switch>
            </div>
        </Router>
    )
}
