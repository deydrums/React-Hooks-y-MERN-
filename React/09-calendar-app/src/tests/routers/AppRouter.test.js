import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';

import { AppRouter } from '../../routers/AppRouter';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth:{
        checking: true,
    }
};
let store = mockStore(initState);
store.dispatch = jest.fn();

describe('Pruebas en <AppRouter />', () => {

    test('Debe de mostrar el loader....', async() => {
        
        const wrapper = mount( 
            <Provider store={ store }>
                    <AppRouter /> 
            </Provider>

        );
        
        expect(wrapper.find('.page-loader').exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();

    });
    
});
