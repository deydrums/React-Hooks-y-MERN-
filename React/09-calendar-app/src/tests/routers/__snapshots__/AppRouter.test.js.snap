// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar el loader.... 1`] = `
<div
  className="loading"
>
  <div
    className="auth__main"
  >
    <div
      className="auth__loading"
    >
      <div
        className="page-loader"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
